### ISSUE TITLE
Header: Improve accessibility for search suggestions

### ISSUE THREAD

**Issue by gfellerph** [swisspost/design-system#2947]
Search suggestions are not announced to assistive technology (https://jira.post.ch/browse/POP-5348)

**Initial announcement:**
![image](https://github.com/swisspost/design-system/assets/1659006/75f09d72-eeee-430a-a7bf-9e877b2e7740)

**Live region announcement:**
![image](https://github.com/swisspost/design-system/assets/1659006/9ef5f470-24eb-4151-85a1-5a7800de948b)

```[tasklist]
### Tasks
- [ ] Available keyboard interaction should be announced initially
- [ ] When new items are available, the number of items should be announced
```
